pkg/frontend/admin_ovsflows_get.go

```diff
@@ -0,0 +1,173 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func (f *frontend) getOVSFlows(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	// Main logic written in f._getOSVFlow() func
+	b, err := f._getOVSFlows(ctx, r, log)
+	if err != nil {
+		log.Error(err)
+	}
+	reply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getOVSFlows(ctx context.Context, r *http.Request, log *logrus.Entry) ([]byte, error) {
+	vars := mux.Vars(r)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	//Checking if the node mentioned in the query is correct or not.
+	nodeName := r.URL.Query().Get("node")
+	if nodeName == "" {
+		err := api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "Invalid Node Name")
+		return nil, err
+	}
+
+	// Getting OpenShiftCluster Document
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		log.Error(api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"]))
+		return nil, err
+	case err != nil:
+		return nil, err
+	}
+
+	// Getting rest config to create *kubernetes.Clientset
+	restConfig, err := restconfig.RestConfig(f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return nil, err
+	}
+	config, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Checking if the given node exists
+	node, err := config.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Creating a pod which will run as privileged in the given node to execute "$ ovs-ofctl -O OpenFlow13 dump-flows br0" on that node to extract the OVS flows.
+	pod, err := createOVSFlowPod(ctx, node.Name, config, log)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	//  The below loop waits of the phase of the created pod to be Succeeded, if not it times out after 60 seconds and returns error.
+	for i := 0; pod.Status.Phase != corev1.PodSucceeded; i++ {
+		if i == 60 {
+			err = api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeInternalServerError, "", "")
+			return nil, err
+		}
+		pod, err = config.CoreV1().Pods("default").Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+		time.Sleep(time.Second)
+	}
+
+	// The OVS flows would be present in the logs of the created pod.
+	b, err := getPodLogs(ctx, log, config, pod)
+	if err != nil {
+		return nil, err
+	}
+
+	// Clearing the pod after extracting its logs.
+	err = config.CoreV1().Pods("default").Delete(ctx, pod.Name, metav1.DeleteOptions{})
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
```

Comment:
We might want to do it as a `defer` function to increase our chances of cleaning up.

Reply:
Sure .. will make the changes ..